File: wookiist/python/1920.py
import sys; input = sys.stdin.readline
def bs(start, end, target, arr):
    if start > end:
        return -1
    mid = int((start+end)/2)
    if target < arr[mid]:
        return bs(start, mid-1, target, arr)
    elif target > arr[mid]:
        return bs(mid+1, end, target, arr)
    return mid
N = int(input())
A = [int(x) for x in input().split()]
A.sort()
M = int(input())
B = [int(x) for x in input().split()]
for b in B:
    if bs(0, N-1, b, A) == -1:
        print(0)
    else:
        print(1)

File: wookiist/go/1920.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

var (
	w  = bufio.NewWriter(os.Stdout)
	sc = bufio.NewScanner(os.Stdin)
)

var (
	A []int
)

func init() {
	sc.Split(bufio.ScanWords)
}

func main() {
	defer w.Flush()
	N := scanInt()
	A = make([]int, N)
	for i := range A {
		A[i] = scanInt()
	}
	sort.Ints(A)
	M := scanInt()
	for i := 0; i < M; i++ {
		t := scanInt()
		if res := binarySearch(0, N-1, t); res != -1 {
			fmt.Fprintln(w, 1)
		} else {
			fmt.Fprintln(w, 0)
		}
	}
}

func binarySearch(start, end, target int) int {
	if start > end {
		return -1
	}
	mid := (start + end) / 2
	if A[mid] > target {
		return binarySearch(start, mid-1, target)
	} else if A[mid] < target {
		return binarySearch(mid+1, end, target)
	}
	return mid
}

func scanInt() int {
	sc.Scan()
	n, _ := strconv.Atoi(sc.Text())
	return n
}
